### PR TITLE
fix (core): Delay added so spell animation plays on unit death

### PR DESCRIPTION
**Pull request by acidmanifesto**
## Changes Proposed:
-  Unit death not showing full spell animation. This delay of 400ms provides that opportunity for it to not be nerfed.
-  0.4ms delay implemented for animation

## Issues Addressed:

- Closes 

## SOURCE:

Sniff Team
## Tests Performed:

- Builds
- Performs

## How to Test the Changes:

1.Fastest way to test is to use mangle as a feral druid and kill stuff
2. observer full and additional animatino is played now
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -977,7 +977,7 @@ uint32 Unit::DealDamage(Unit* attacker, Unit* victim, uint32 damage, CleanDamage
 
         //if (attacker && victim->GetTypeId() == TYPEID_PLAYER && victim != attacker)
         //victim->ToPlayer()->UpdateAchievementCriteria(ACHIEVEMENT_CRITERIA_TYPE_TOTAL_DAMAGE_RECEIVED, health); // pussywizard: optimization
-
+        std::this_thread::sleep_for(400ms);
```
Comment (Nefertumm) [inline src/server/game/Entities/Unit/Unit.cpp:980]: Add a little comment as the other PR 👍 
All good

**Review by Nefertumm — APPROVED** [review]

**Review by Winfidonarleyan — APPROVED** [review]

File: src/server/game/Entities/Unit/Unit.cpp (hunk 1 above)
Comment (acidmanifesto) [reply src/server/game/Entities/Unit/Unit.cpp:983]: added

**Comment by Chero** [comment]
Is this PR a joke?

If you kill 50 mobs at once with an AoE, you will freeze the server for 20 seconds.

**Comment by GrenderG** [comment]
Hello! I don't think this is a good approach to solve the issue.

**Comment by Nyeriah** [comment]
> I don't think this is a good approach to solve the issue.

Hey!

Thanks for your feedback, but this was already reverted. I'll lock this PR now because it seems to be misleading too many people.

For reference, it was reverted here: https://github.com/azerothcore/azerothcore-wotlk/commit/8f249a3d84c9086711dd96185aa28334a43eeee8

If you are encountering issues, please update your core.

A new solution to this issue will be presented soon, make sure to check it out once it's ready.

Thanks once again!

**Pull request merged**